### PR TITLE
Dennis_Batiste_Regression__Part3

### PR DESCRIPTION
**Pull request by dennibat**

### CODE REVIEW

**Comment by llpk79** [comment]
### Assignment
- [x] Look at the data. Choose a feature, and plot its relationship with the target.
    - No reason to re-import the data after scrubbing the outliers. Remove that line and the rest of the notebook looks a lot more meaningful, as you aptly predicted :). Now, check out the code in the third cell of the notebook for the answer to your question of how the outlier can be scrubbed. 
- [x] Use scikit-learn for linear regression with one feature. You can follow the 5-step process from Jake VanderPlas.
   - Nice job. Your results will be better with the outliers gone.
- [x] Define a function to make new predictions and explain the model coefficient.
    - NIce!
    - How can you make this function more versatile? What if we want to know about bathrooms or some other feature?
- [ ] Organize and comment your code.
    - Can use some work here. Pay close attention to things like importing twice. Slow down and read the code given to you to make sure you know what has been done to the data you're being given. You'll need to do those things on your own on the job.
### Stretch Goals
- [ ] Do linear regression with two or more features.
    - Give this a shot!!!

**Comment by dennibat** [comment]
Late and still not done yet

**Comment by llpk79** [comment]
## Regression 2
#### Make sure to get rid of that extra `df = pd.read_csv(...)` that's negating all the work above it!
- [x] Do train/test split. Use data from April & May 2016 to train. Use data from June 2016 to test.
    - Look at the dates you have in the dataset. Seems like you can split the data around one data.
- [ ] Engineer at least two new features. (See below for explanation & ideas.)
    - 😢 
- [x] Fit a linear regression model with at least two features.
    - Technically done.
- [ ] Get the model's coefficients and intercept.
    - 😢 
- [ ] Get regression metrics RMSE, MAE, and R^2, for both the train and test data.
    - 😢 
- [ ] What's the best test MAE you can get? Share your score and features used with your cohort on Slack!
    - 😢 
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️ 

Make sure to spend time finishing the basic assignment with your own code before playing with fancy graphing functions.

**Comment by llpk79** [comment]
## Ridge Regression
- [x] Do train/test split. Use data from January — March 2019 to train. Use data from April 2019 to test.
- [x] Do one-hot encoding of categorical features.
    - Kinda strange that `Land_SQUARE_FEET` is in the categorical features.
- [x] Do feature selection with `SelectKBest`.
    - `k = 'all'`. What's the point of using a selection algorithm if we don't select anything? 
    - From the docs :`The “all” option bypasses selection`. 
    - Make `k` a number of features to select.
- [ ] Do [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html).
- [x] Fit a ridge regression model with multiple features.
- [x] Get mean absolute error for the test set.
    - Getting things done in the fewest lines of code is not a very good goal. Writing code that is readable by others is much more important. 
- [x] As always, commit your notebook to your fork of the GitHub repo.
    - ✔️
